src/interpreter.js

```diff
@@ -155,57 +156,37 @@ const compileExpression = (expression) => {
 
 
 const compileLiteral = (literal) => {
-  if(literal.nodeType === 'NullLiteral') {
-    return `null`
+  const rules = {
+    NullLiteral: () => 'null',
+    SelfLiteral: () => 'this',
+    BooleanLiteral: ({ value }) => `${value}`,
+    NumberLiteral: ({ value }) => `${value}`,
+    StringLiteral: ({ value }) => `"${value}"`,
+    SetLiteral: ({ values }) => `new Set([ ${values.map(compileExpression).join()} ])`,
+    ListLiteral: ({ values }) => `[ ${values.map(compileExpression).join()} ]`
   }
-
-  if(literal.nodeType === 'SelfLiteral') {
-    return `this`
+  if (rules[literal.nodeType]) {
```

Comment:
@javierfernandes I was reading last night how Babel does this kind of things and got the idea for writing something very similar.

My proposal is a bit different, let's relace all compile functions with a "visitor" object and add the visitor methods named for the nodetypes like you did as attributes:

```
    const compile = {
       NumberLiteral({ value }) { return `${value}` },
      BinaryOp({ op, left, right }) { return ... },
     ...
    }
```

We can also add a "generic"  compile method, so it can be called on any node
```    compile.call = (node) => call[node.nodeType](node) ```